lte/gateway/python/magma/mobilityd/ipv6_allocator_pool.py

```diff
@@ -0,0 +1,191 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+import random
+
+from ipaddress import ip_address, ip_network
+from typing import List, Optional
+
+from .ip_descriptor import IPDesc, IPType
+from .ip_allocator_base import IPAllocator, OverlappedIPBlocksError, IPNotInUseError
+from .ip_descriptor import IPv6SessionAllocType
+
+IPV6_PREFIX_PART_LEN = 64
+IID_PART_LEN = 64
+MAX_IPV6_CONF_PREFIX_LEN = 48
+MAX_CALC_TRIES = 5
+
+
+class IPv6AllocatorPool(IPAllocator):
+    def __init__(self, config):
+        super().__init__()
+        self._config = config
+        self._assigned_ip_block = None
+        self._allocated_iid = set()
+        self._sid_session_prefix_allocated = {}
+        self._ipv6_session_prefix_alloc_mode = self._config[
+            'ipv6_ip_allocator_type']
+
+    def add_ip_block(self, ipblock: ip_network):
+        """
+        Adds IP block to the assigned IP block of the IPv6 allocator
+        :param ipblock: IPv6 ip_network object
+        """
+        if self._assigned_ip_block and ipblock.overlaps(
+                self._assigned_ip_block):
+            logging.error("Overlapped IP block: %s", ipblock)
+            raise OverlappedIPBlocksError(ipblock)
+
+        if ipblock.prefixlen > MAX_IPV6_CONF_PREFIX_LEN:
+            msg = "IPv6 block exceeds maximum allowed prefix length"
+            logging.error(msg)
+            raise InvalidIPv6NetworkError(msg)
+
+        # For now only one IPv6 network is supported
+        self._assigned_ip_block = ipblock
+
+    def remove_ip_blocks(self, *ipblocks: List[ip_network],
+                         force: bool = False) -> List[ip_network]:
+        """
+        Removes assigned IP block (as it only supports one for now)
+        :param ipblocks: List of ip_network objects to remove
+        :param force:
+        :return: Removed list of IP blocks
+        """
+        removed_blocks = []
+        if self._assigned_ip_block in ipblocks:
```

Comment:
think this would be a tad clearer if you inverted the conditional and returned early